src/HsmMoments.cc

```diff
@@ -145,15 +146,56 @@ void HsmPsfMomentsAlgorithm::measure(
     geom::Point2D center = _centroidExtractor(source, _flagHandler);
 
     typedef afw::image::Mask<afw::image::MaskPixel> Mask;
-    PTR(afw::detection::Psf::Image) image = exposure.getPsf()->computeKernelImage(center);
+    PTR(afw::detection::Psf::Image) psfImage;
+
+    if (_ctrl.debias) {
+        typedef afw::detection::Psf::Pixel PsfPixel;
+
+        // draw Psf with same offset as star
+        psfImage = exposure.getPsf()->computeImage(center);
```

Comment:
> So, for now, can we put this behind a config option, in addition to adjusting the center below?

I'm happy to put these behind a config option, but do you have a suggestion for how to interpolate the variance plane in the case where `computeKernelImage` is used?  Nearest pixel is easy enough, but maybe there's code somewhere else in the stack that could do something better?

Reply:
> Nearest pixel is easy enough, but maybe there's code somewhere else in the stack that could do something better?

Good point; I hadn't thought about what to do about the variance plane when proposing this.  There are better interpolants, but I'm not super familiar with it, and it'll bring some complications with it (e.g. edge effects).  Let's just do nearest pixel for now and document that as a limitation of that config choice.

Reply:
> This (drawing PSF with the same offset as the star) is an interesting choice. It might be the right one, but I think that's an experimental question (if they're not equivalent, it says something about our interpolation that would matter in other places).

Looking into this a bit more.  I decided it would be worth trying out drawing an offset PSF even in the no-additional-noise-added case, and have found that, indeed, the current unit tests fail in this case.  Specifically, getting the HSM _centroid_ of a GaussianPsf that's drawn off-center still works fine, but the HSM _second moments_ are affected.  For a half-pixel shift with expected Ixx=4.0 pix^2, HSM yields 3.988, so I'd need to reduce the test tolerance from 3 decimals to 1.  (For a centered PSF image, HSM actually gives the expected result to 5 decimals).

I suspect this says more about HSM though than about PSF interpolation.  But it definitely seems like a source of bias when comparing PSFs and stars.

Reply:
> I suspect this says more about HSM though than about PSF interpolation. 

I take it back, as when I try to reproduce this using just GalSim, HSM seems perfectly able to deal with pixel offsets.  Is `GaussianPsf.computeImage()` implemented by direct computation or by Lanczosing the `computeKernelImage()` result?  If it's the latter, then yeah, that could be a problem.  The next question is then what do `Psfex` and `Piff` do?  Piff at least I know draws the offset image directly (though there is certainly interpolation under the hood, it's fundamentally part of the model, not something added on top).

```python
gaussian = galsim.Gaussian(sigma=2.0)

img = gaussian.drawImage(nx=35, ny=35, scale=1.0, method='no_pixel')
mom = galsim.hsm.FindAdaptiveMom(img)
print(mom.moments_centroid)
print(mom.moments_sigma)
print(mom.observed_shape)

# draw offset directly
img2 = gaussian.drawImage(nx=35, ny=35, scale=1.0, method='no_pixel', offset=(0.5, 0.0))
mom2 = galsim.hsm.FindAdaptiveMom(img2)
print()
print(mom2.moments_centroid)
print(mom2.moments_sigma)
print(mom2.observed_shape)

# Use Lanczos magic to draw with offset
ii = galsim.InterpolatedImage(img, scale=1.0, normalization='sb', x_interpolant='Lanczos3')
img3 = ii.drawImage(nx=35, ny=35, scale=1.0, method='no_pixel', offset=(0.5, 0.0))
mom3 = galsim.hsm.FindAdaptiveMom(img3)
print()
print(mom3.moments_centroid)
print(mom3.moments_sigma)
print(mom3.observed_shape)
```

yields

```
galsim.PositionD(18.0,18.0)
2.0
galsim.Shear(g1=5.551115123125783e-17,g2=9.721081979824222e-18)

galsim.PositionD(18.5,18.0)
2.0
galsim.Shear(g1=-6.354167192412774e-10,g2=1.1135718818786897e-17)

galsim.PositionD(18.5,18.0)
1.997077465057373
galsim.Shear(g1=-0.0014623346344376938,g2=1.5930238952803163e-17)
```

Reply:
I'm pretty sure `CoaddPsf` is just going to use a Lanczos kernel to interpolate from `computeKernelImage` to `computeImage`, and yes, I've been worried about that as a potential problem for a while - though I _think_ someone tried the experiment of upping the Lanczos order there at some point and found that didn't explain the difference between our single-visit rho statistics and coadd rho statistics.  We should definitely return to that question after we have a consistent sample of reserved stars.

PSFEx would do its own interpolation, and I think that entails a Lanczos interpolation of the upsampled model image, but I'm not certain.

Reply:
Sounds good.  

I think the immediate question is: should I fix `afwDetection::GaussianPsf.computeImage()` now to not use Lanczos so that new HSM unit tests pass with tight thresholds, or should I just use really loose thresholds on those new tests?  I opted for loose thresholds for now, but I don't think it'd actually be that much work to fixup GaussianPsf.

Reply:
I think the loose thresholds are fine here, and I suspect modifying `GaussianPsf` might would perturb a bunch of existing tests in a way that would be just a waste of time to deal with.

Reply:
I found a third option here, which was to implement a pure-python Gaussian Psf that handles `computeImage` exactly.  Thanks to trampolines this is only ~40 lines of pure python now :)